misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathAssertionsSpec.kt

```diff
@@ -672,6 +658,169 @@ abstract class PathAssertionsSpec(
         }
     }
 
+    describeFun(isExecutable) {
+        val isExecutableFun = isExecutable.lambda
+        val expectedMessage = "$isDescr: ${EXECUTABLE.getDefault()}"
+
+        context("not accessible") {
+            it("throws an AssertionError for non-existent path") withAndWithoutSymlink { maybeLink ->
+                val file = maybeLink.create(tempFolder.tmpDir.resolve("nonExistent"))
+                expect {
+                    expect(file).isExecutableFun()
+                }.toThrow<AssertionError>().message {
+                    contains(expectedMessage, FAILURE_DUE_TO_NO_SUCH_FILE.getDefault())
+                    containsExplanationFor(maybeLink)
+                }
+            }
+
+            itPrintsFileAccessProblemDetails { testFile ->
+                expect(testFile).isExecutableFun()
+            }
+        }
+
+        context("POSIX: executable", skip = ifPosixNotSupported) {
+            it("does not throw for file") withAndWithoutSymlink { maybeLink ->
+                val file = maybeLink.create(tempFolder.newFile("executable"))
+                file.whileWithPermissions(OWNER_READ, OWNER_WRITE, OWNER_EXECUTE) {
+                    expect(file).isExecutableFun()
+                }
+            }
+
+            it("does not throw for directory") withAndWithoutSymlink { maybeLink ->
+                val folder = maybeLink.create(tempFolder.newDirectory("executable"))
+                folder.whileWithPermissions(OWNER_READ, OWNER_WRITE, OWNER_EXECUTE) {
+                    expect(folder).isExecutableFun()
+                }
+            }
+        }
+
+        context("ACL: executable", skip = ifAclNotSupported) {
+            it("does not throw for file") withAndWithoutSymlink { maybeLink ->
+                val file = maybeLink.create(tempFolder.newFile("executable"))
+                file.whileWithAcl(TestAcls::all) {
+                    expect(file).isExecutableFun()
+                }
+            }
+
+            it("does not throw for directory") withAndWithoutSymlink { maybeLink ->
+                val folder = maybeLink.create(tempFolder.newDirectory("executable"))
+                folder.whileWithAcl(TestAcls::all) {
+                    expect(folder).isExecutableFun()
+                }
+            }
+        }
+
+        context("POSIX: not executable", skip = ifPosixNotSupported) {
+            val expectedPermissionHint = String.format(HINT_ACTUAL_POSIX_PERMISSIONS.getDefault(), "u=rw g=r o=")
+
+            it("throws an AssertionError for a file") withAndWithoutSymlink { maybeLink ->
+                val expectedTypeHint = String.format(
+                    FAILURE_DUE_TO_PERMISSION_FILE_TYPE_HINT.getDefault(),
+                    A_FILE.getDefault(),
+                    EXECUTABLE.getDefault()
```

Comment:
maybe worth introducing two functions, this is repeated several times
- fileTypeHintFor(translatable: Translatable)
- directoryTypeHintFor(translatable: Translatable)

Reply:
Done